### PR TITLE
Add custom css and js to singles and pages

### PR DESCRIPTION
**Pull request by FrancisMawn**

### CODE REVIEW

File: config/general.php [hunk 1]
```diff
@@ -30,8 +30,16 @@
         // Path to error templates
         'errorTemplatePrefix' => '_errors/',
 
-        // Allow XML file upload in assets
-        'extraAllowedFileExtensions' => 'xml',
+        // Allow extra file types to be uploaded in assets
+        'extraAllowedFileExtensions' => ['xml', 'css'],
```
Comment (HopSocial) [inline config/general.php]: Pas besoin d'ajouter JS ici?

Comment (jlcharette) [inline config/general.php]: Is JS part of the default settings?

File: config/general.php [hunk 2]
```diff
@@ -30,8 +30,16 @@
...
+
+        // List of additional file types Craft should support
+        'extraFileKinds' => [
```
Comment (jlcharette) [inline config/general.php]: What is `extraFileKinds` vs `extraAllowedFileExtensions`?

File: templates/index.twig [hunk 3]
```diff
@@ -6,10 +6,18 @@
 
 {% extends '_layouts/default' %}
 
+{% block styles %}
+    {% include '_shared/styles' %}
```
Comment (jlcharette) [inline templates/index.twig]: Why here rather than in `_layouts/default`

File: templates/index.twig [hunk 4]
```diff
@@ -6,10 +6,18 @@
...
 {% block page %}
     {% include [
         '_entries/' ~ entry.section.handle ~ '/' ~ entry.type,
         '_entries/' ~ entry.section.handle ~ '/default',
         '_entries/default'
     ] %}
 {% endblock %}
+
+{% block scripts %}
+    {% include '_shared/scripts' %}
```
Comment (jlcharette) [inline templates/index.twig]: Why here rather than in `_layouts/default`

File: config/general.php (hunk 1 above)
Comment (FrancisMawn) [reply config/general.php]: Yes

File: config/general.php (hunk 2 above)
Comment (FrancisMawn) [reply config/general.php]: _extraFileKinds_ defines a new file type for when you setup the field. 
_extraAllowedFileExtensions_ makes it available as a choice.

File: templates/index.twig (hunk 4 above)
Comment (FrancisMawn) [reply templates/index.twig]: My reasoning was that the **default layout** defines blocks where css and scripts should be placed in the page. The same way it has a page block to define where the page content should be placed.

The **index.twig** which extends the **default layout** has its own logic to handle that.

So we could technically have another template file that also extends the **default layout** which could have its own logic to define extra javascript/css.

Comment (jlcharette) [reply templates/index.twig]: Fair enough. But the default layout will no longer include the WET-BOEW. Every template extending the default layout will require the css/js logic.

Comment (FrancisMawn) [reply templates/index.twig]: That's a good point. It's a bit of a trade off I guess. The default layout is now more flexible in the sens that we could have custom pages that do not use the WET-BOEW library.

Do you want me to move them back in the default layout?

**Pull request merged**